src/main/java/de/tum/in/www1/artemis/config/migration/entries/MigrationEntry20220619_123000.java

```diff
@@ -0,0 +1,145 @@
+
+import de.tum.in.www1.artemis.config.migration.MigrationEntry;
+import de.tum.in.www1.artemis.domain.ProgrammingExercise;
+import de.tum.in.www1.artemis.domain.Result;
+import de.tum.in.www1.artemis.repository.ProgrammingExerciseRepository;
+import de.tum.in.www1.artemis.repository.ResultRepository;
+
+/**
+ * This migration extracts the information about a programming result and stores it inside the new variables so the result String can still be displayed correctly in the client
+ */
+@Component
+public class MigrationEntry20220619_123000 extends MigrationEntry {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(MigrationEntry20220619_123000.class);
+
+    private final ResultRepository resultRepository;
+
+    private final ProgrammingExerciseRepository programmingExerciseRepository;
+
+    public MigrationEntry20220619_123000(ResultRepository resultRepository, ProgrammingExerciseRepository programmingExerciseRepository) {
+        this.resultRepository = resultRepository;
+        this.programmingExerciseRepository = programmingExerciseRepository;
+    }
+
+    /**
+     * Retrieves all results from the database and executes the processing method in batches of 100 results to prevent database timeouts
+     */
+    @Override
+    public void execute() {
+        Field resultString;
+        try {
+            resultString = Result.class.getDeclaredField("resultString");
+            resultString.setAccessible(true);
+        }
+        catch (NoSuchFieldException e) {
+            // The migration was already successful and the deprecated field is removed
+            return;
+        }
+
+        List<ProgrammingExercise> programmingExercises = programmingExerciseRepository.findAll();
+        LOGGER.info("Found {} programming exercise to process.", programmingExercises.size());
+        long numberOfUnprocessedResults = 0;
+        ZonedDateTime latestUnprocessedResultDate = null;
+
+        for (ProgrammingExercise exercise : programmingExercises) {
+            List<Result> results = resultRepository.findAllByExerciseId(exercise.getId());
+            results.removeIf(result -> getResultString(result, resultString) == null);
+            LOGGER.info("Found {} results for the exercise with ID {} to process.", results.size(), exercise.getId());
+
+            for (List<Result> resultList : Lists.partition(results, 100)) {
+                LOGGER.info("Process (next) 100 results for the migration in one batch...");
+                List<Result> unprocessedResults = processProgrammingResults(resultList, resultString);
+                numberOfUnprocessedResults += unprocessedResults.size();
+                latestUnprocessedResultDate = (Stream.concat(unprocessedResults.stream().map(Result::getCompletionDate), Stream.of(latestUnprocessedResultDate)))
+                        .filter(Objects::nonNull).max(Comparator.naturalOrder()).orElse(null);
+            }
+        }
+
+        if (numberOfUnprocessedResults != 0) {
```

Comment:
You just closed that without change or comment.

Reply:
This change is not necessary in my opinion since it either does exactly the same or my variant is even safer if we had an integer overflow in numberOfUnprocessedResults